crypto/dsa/dsa_ameth.c

```diff
@@ -275,55 +279,34 @@ static int dsa_missing_parameters(const EVP_PKEY *pkey)
 {
     DSA *dsa;
     dsa = pkey->pkey.dsa;
-    if (dsa == NULL || dsa->p == NULL || dsa->q == NULL || dsa->g == NULL)
-        return 1;
-    return 0;
+    return dsa == NULL
+           || dsa->params.p == NULL
+           || dsa->params.q == NULL
+           || dsa->params.g == NULL;
```

Comment:
Style nit: a bit too much indentation (more than 4 spaces)

A nice property with 4 space indentation here is that the subexpressions will align exactly :-)